tests/main/snap-seccomp/task.yaml

```diff
@@ -0,0 +1,98 @@
+
+execute: |
+    if [ "$(snap debug confinement)" != strict ] ; then
+        echo "SKIP: these tests are only useful on systems with working seccomp"
+        exit 0
+    fi
+
+    echo "Install test-snapd-tools and verify it works"
+    snap install test-snapd-tools
+    test-snapd-tools.echo hello | MATCH hello
+
+    # from the old test_complain
+    echo "Test that the @complain keyword works"
+    rm -f ${PROFILE}.bpf
+    cat >"${PROFILE}.in" <<EOF
+    # some comment
+    @complain
+    EOF
+    $SNAP_SECCOMP compile ${PROFILE}.in ${PROFILE}.bpf
+    echo "Ensure the code still runs"
+    test-snapd-tools.echo hello | MATCH hello
+
+    # from the old test_complain_missed
+    rm -f ${PROFILE}.bpf
+    cat >"${PROFILE}.in" <<EOF
+    # super strict filter
+    @complai
+    @complaim
+    @omplain
+    @COMPLAIN
+    complain
+    EOF
+    $SNAP_SECCOMP compile ${PROFILE}.in ${PROFILE}.bpf
+    echo "Ensure the code cannot not run due to impossible filtering"
+    if test-snapd-tools.echo hello; then
+        echo "filtering broken: program should have failed to run: program should have failed to run"
+        exit 1
+    fi
+    
+    # from the old test_unrestricted
+    echo "Test that the @unrestricted keyword works"
+    rm -f ${PROFILE}.bpf
+    cat >"${PROFILE}.in" <<EOF
+    # some comment
+    @unrestricted
+    EOF
+    $SNAP_SECCOMP compile ${PROFILE}.in ${PROFILE}.bpf
+    echo "Ensure the code still runs"
+    test-snapd-tools.echo hello | MATCH hello
+
+    # from the old test_unrestricted_missed
+    rm -f ${PROFILE}.bpf
+    cat >"${PROFILE}.in" <<EOF
+    # super strict filter
+    @unrestricte
+    @unrestrictes
+    @nrestricted
+    @UNRESTRICTED
+    unrestricted
+    EOF
+    $SNAP_SECCOMP compile ${PROFILE}.in ${PROFILE}.bpf
+    echo "Ensure the code cannot not run due to impossible filtering"
+    if test-snapd-tools.echo hello; then
+        echo "filtering broken: program should have failed to run"
+        exit 1
+    fi
+
+    # from the old test_noprofile
+    rm -f ${PROFILE}.bpf
+    echo "Ensure the code cannot not run due to missing filter"
+    if test-snapd-tools.echo hello; then
+        echo "filtering broken: program should have failed to run"
+        exit 1
+    fi
+
+    echo "Break snapd.test-snapd-tools.bpf to ensure (kernel) validation works"
+    dd if=/dev/urandom of=${PROFILE}.bpf count=1 bs=1024
+    if output=$(test-snapd-tools.echo hello 2>&1 ); then
+        echo "test-snapd-tools.echo should fail with invalid seccomp profile"
+        exit 1
+    fi
+    echo $output | MATCH "prctl.*Invalid argument"
+
+    echo "Add huge snapd.test-snapd-tools.bpf to ensure size limit works"
+    dd if=/dev/zero of=${PROFILE}.bpf count=50 bs=1M
+    if output=$(test-snapd-tools.echo hello 2>&1 ); then
+        echo "test-snapd-tools.echo should fail with big seccomp profile"
+        exit 1
+    fi
+    echo $output | MATCH "profile .* exceeds .* bytes"
+
```

Comment:
Can you add additional tests for:
* profile is missing, you run snap-seccomp and therefore snap-seccomp fails
* profile is empty, you run snap-seccomp and therefore the snap fails to start (default closed/restrictive policy)